app/models/piece.rb

```diff
@@ -45,14 +45,17 @@ def move_to!(new_x, new_y)
     end
   end
 
-  def is_obstructed?(x1, y1, x2, y2)
-    vertical_move = x1 === x2
+  def is_obstructed?(new_x, new_y)
+    x1 = self.x_position
+    y1 = self.y_position
+
+    vertical_move = x1 === new_x
```

Comment:
@Nikkovichert would be great to pull out a few methods so that the is_obstructed would look sth like:

```
def is_obstructed?(new_x, new_y)
  is_vertically_obstructed?(new_x, new_y) || is_horizontally_obstructed?(new_x, new_y)
end
```

so that the logic is easier to follow!

Reply:
sure, I believe Linh was saying something about that a while ago too, I'll get on it now